tests/utils/stl/test/format.py

```diff
@@ -10,312 +10,318 @@
-        for path in source_dir.iterdir():
-            if path.is_file() and path.name.endswith('.dat'):
-                shutil.copy2(path, exec_dir / path.name)
-
-    def cleanup(self, test):
-        shutil.rmtree(test.getExecDir(), ignore_errors=True)
-        shutil.rmtree(test.getOutputDir(), ignore_errors=True)
-
-    def getIntegratedScriptResult(self, test, lit_config):
-        if test.skipped:
-            return (lit.Test.SKIPPED, "Test was marked as skipped")
-
-        name = test.path_in_suite[-1]
-        name_root, name_ext = os.path.splitext(name)
-        is_sh_test = name_root.endswith('.sh')
-        is_objcxx_test = name.endswith('.mm')
-
-        if is_sh_test:
-            return (lit.Test.UNSUPPORTED,
-                    "Sh tests are currently unsupported")
-
-        if is_objcxx_test:
-            return (lit.Test.UNSUPPORTED,
-                    "Objective-C tests are unsupported")
-
-        if test.config.unsupported:
-            return (lit.Test.UNSUPPORTED,
-                    "A lit.local.cfg marked this unsupported")
-
-        if lit_config.noExecute:
-            return lit.Test.Result(lit.Test.PASS)
-
-        if test.expected_result is None:
-            script = lit.TestRunner.parseIntegratedTestScript(
-                test, require_script=False)
-
-            if isinstance(script, lit.Test.Result):
-                return script
-
-        return None
-
-    def execute(self, test, lit_config):
-        result = None
-        while True:
+                    for envEntry, envNum in zip(envEntries, itertools.count()):
+
+                        yield testClass(testSuite, pathInSuite + (filename,),
+                                        litConfig, localConfig, envEntry,
+                                        formatString.format(envNum))
+
+    def _parseScript(self, test):
+        additionalCompileFlags = []
+        fileDependencies = []
+        parsers = [
+            lit.TestRunner.IntegratedTestKeywordParser('FILE_DEPENDENCIES:',
+                                                       lit.TestRunner.ParserKind.LIST,
+                                                       initial_value=fileDependencies),
+            lit.TestRunner.IntegratedTestKeywordParser('ADDITIONAL_COMPILE_FLAGS:',
+                                                       lit.TestRunner.ParserKind.LIST,
+                                                       initial_value=additionalCompileFlags)
+        ]
+
+        script = \
+            lit.TestRunner.parseIntegratedTestScript(test, additional_parsers=parsers, require_script=False)
+
+        if isinstance(script, lit.Test.Result):
+            test.result = script
+            return
+
+        test.compileFlags.extend(additionalCompileFlags)
+        test.fileDependencies.extend(fileDependencies)
+
+    def _handleIsenseRspFile(self, test, litConfig):
+        if litConfig.edg_drop is not None:
+            with open(isenseRspPath) as f:

```

Comment:
It doesn't look like `isenseRspPath` is ever defined here- the callers use a variable with the same name to add the `--write-isense-rsp` flag, but only in an `if` scope so it never reaches here.

Reply:
A lot of output gets (unhelpfully) swallowed by lit. Give running the tests with the `-a` option a shot.

Reply:
Ahh, thanks- looks like `_handleIsenseRspFile` is getting skipped because the `if TestType.COMPILE in test.testType` branch is using `return` instead of `yield`.

Once I patch that to a `yield` and forward `isenseRspPath` to `_handleIsenseRspFile`, the next problem is that `_handleIsenseRspFile`'s `yield TestStep` line doesn't have `shared` in scope for `shared.execDir`. (Incidentally, the exception handler in `execute` is also failing to report this, because it still uses `lit_config` instead of `litConfig`.)

Once I also forward `shared` to `_handleIsenseRspFile` it seems to work, though I haven't looked at the custom formats that also call `_handleIsenseRspFile`.

Reply:
Alright I switched it up a bit and I think it should work now.
The only question is around multi-TU tests though I don't think this was working for them in the first place.